tests/infra_test.go

```diff
@@ -95,6 +99,83 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
 		}
 	})
 
+	Describe("changes to the kubernetes client", func() {
+
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+		})
+
+		It("on the controller rate limiter should lead to delayed VMI starts", func() {
+			By("first getting the basetime for a replicaset")
+			replicaset := tests.NewRandomReplicaSetFromVMI(libvmi.NewCirros(libvmi.WithResourceMemory("1Mi")), int32(0))
+			replicaset, err = virtClient.ReplicaSet(util.NamespaceTestDefault).Create(replicaset)
+			Expect(err).ToNot(HaveOccurred())
+			start := time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			fastDuration := time.Now().Sub(start)
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 0)
+
+			By("reducing the throughput on controller")
+			originalKubeVirt := util.GetCurrentKv(virtClient)
+			originalKubeVirt.Spec.Configuration.ControllerConfiguration = &v1.ReloadableComponentConfiguration{
+				RestClient: &v1.RESTClientConfiguration{
+					RateLimiter: &v1.RateLimiter{
+						TokenBucketRateLimiter: &v1.TokenBucketRateLimiter{
+							Burst: 3,
+							QPS:   2,
+						},
+					},
+				},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+			By("starting a replicaset with reduced throughput")
+			start = time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			slowDuration := time.Now().Sub(start)
+			Expect(slowDuration.Seconds()).To(BeNumerically(">", 5*fastDuration.Seconds()))
+		})
+
+		It("on the virt handler rate limiter should lead to delayed VMI running states", func() {
+			By("first getting the basetime for a replicaset")
+			targetNode := util.GetAllSchedulableNodes(virtClient).Items[0]
+			vmi := libvmi.NewCirros(
+				libvmi.WithResourceMemory("1Mi"),
+				libvmi.WithNodeSelectorFor(&targetNode),
```

Comment:
Why do you need to force the placement decision?

Reply:
Otherwise the VMs get distributed to differend virt-handlers which will make the rate-limit throttling less predictable.